src/v3/src/transformer/oktaVerify/transformOktaVerifyChannelSelection.ts

```diff
@@ -62,11 +62,19 @@ export const transformOktaVerifyChannelSelection: IdxStepTransformer = ({
     'authenticator.channel',
     formBag.uischema.elements as UISchemaElement[],
   ) as FieldElement;
-  channelSelectionElement.label = loc('oie.enroll.okta_verify.select.channel.description', 'login');
+  channelSelectionElement.label = loc('oie.enroll.okta_verify.select.channel.subtitle', 'login');
   channelSelectionElement.options.format = 'radio';
   const { options: { inputMeta: { options = [] } } } = channelSelectionElement;
   channelSelectionElement.options.customOptions = options
-    .filter((opt) => opt.value !== lastSelectedChannel)
+    .filter((opt) => {
+      if (opt.value === 'samedevice') {
+        // filter out the samedevice channel so it is not displayed as a radio
+        // and set flag to true so we know to add it as a link instead
+        sameDeviceChannelAvailable = true;
```

Comment:
Is there a way to avoid using a stateful flag for this? This makes it more difficult to understand the code paths that will be taken, especially when the conditional reading this value is farther down. Even if that means iterating through the `options` a second time it is probably worth it since N is going to be small.

Reply:
yeah I should be able to refactor this to not use a state flag